core/src/main/java/com/yahoo/imapnio/async/request/ListStatusCommand.java

```diff
@@ -0,0 +1,129 @@
+package com.yahoo.imapnio.async.request;
+
+import java.nio.charset.StandardCharsets;
+
+import javax.annotation.Nonnull;
+
+import com.sun.mail.imap.protocol.BASE64MailboxEncoder;
+import com.yahoo.imapnio.async.exception.ImapAsyncClientException;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+/**
+ * This class defines IMAP LIST-STATUS Command, RFC5819. This extension is part of LIST-EXTENDED extension, RFC5258.
+ *
+ * This command is specialized to support LIST-STATUS. It does not support list-select-opts, nor other RETURN options.
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc5819#page-2" target="_blank">LIST-STATUS extension</a>
+ * @see <a href="https://tools.ietf.org/html/rfc5258#page-19" target="_blank">LIST-EXTENDED extension</a>
+ *
+ *      <blockquote>
+ *
+ *      <pre>
+ *
+ * list =      "LIST" [SP list-select-opts] SP mailbox SP mbox-or-pat
+ *             [SP list-return-opts]
+ *
+ * return-option =/ status-option
+ *
+ * status-option = "STATUS" SP "(" status-att *(SP status-att) ")"
+ *                 ;; This ABNF production complies with
+ *                 ;; &lt;option-extension&gt; syntax.
+ *
+ *      </pre>
+ *
+ *      </blockquote>
+ */
+public class ListStatusCommand extends ImapRequestAdapter {
+
+    /** Byte array for CR and LF, keeping the array local so it cannot be modified by others. */
+    private static final byte[] CRLF_B = { '\r', '\n' };
+
+    /** Double right parenthesis. */
+    private static final byte[] DOUBLE_RP_B = { ')', ')' };
+
+    /** List command name. */
+    private static final String LIST_SP = "LIST ";
+
+    /** Byte array for above command. */
+    private static final byte[] LIST_SP_B = LIST_SP.getBytes(StandardCharsets.US_ASCII);
+
+    /** RETURN literal and Left Parenthesis. */
+    private static final String RETURN_STATUS_LP = "RETURN (STATUS (";
+
+    /** Byte array for above. */
+    private static final byte[] RETURN_STATUS_LP_B = RETURN_STATUS_LP.getBytes(StandardCharsets.US_ASCII);
+
+    /** Reference name. */
+    private String ref;
+
+    /** Mailbox(s) pattern. */
+    private String pattern;
+
+    /** Status data item names. */
+    private String[] items;
+
+    /**
+     * Initializes a {@link ListStatusCommand}.
+     *
+     * @param ref the reference string
+     * @param pattern folder name with possible wildcards, see RFC3501 list command for detail. Note that this pattern conforms to RFC3501 LIST
+     *            command pattern.
+     * @param items list of items. Available ones are : "MESSAGES", "RECENT", "UNSEEN", "UIDNEXT", "UIDVALIDITY"
```

Comment:
Items list here and  in the ListStatusCommandTest are not the same.

Reply:
Done. Fixed by not having this in documentation to avoid misleading.   UT has various combinations.